### PR TITLE
ESLint: Enable and enforce remaining i18n rules for the plugin (e.g. no trailing spaces)

### PR DESCRIPTION
**Pull request by swissspidy**
## What?

These two ESLint rules are available in the ESLint plugin but are not enabled by default. This enables them for the Gutenberg repository.

Includes changes from #59835 to remove trailing spaces (props @maheshbohara)

## Why?

Because GB should follow i18n best practices :-)

## Testing Instructions

- Components now using `createInterpolateElement` (e.g. the welcome guide) should still render with the proper link
- Test should pass

### CODE REVIEW

**Comment by gziolo** [comment]
CI reports:

```
328 problems (10 errors, 318 warnings)
10 errors and 0 warnings potentially fixable with the `--fix` option.
```

These warnings are unrelated but errors can be easily fixed.

**Comment by swissspidy** [comment]
#59835 should fix many of these errors

File: packages/block-library/src/post-navigation-link/edit.js [hunk 1]
```diff
@@ -49,8 +49,8 @@ export default function PostNavigationLinkEdit( {
 	const displayArrow = arrowMap[ arrow ];
 
 	if ( showTitle ) {
-		/* translators: Label before for next and previous post. There is a space after the colon. */
```
Comment (gziolo) [inline packages/block-library/src/post-navigation-link/edit.js:52]: It looks like it was intentional to include the space.

**Review by gziolo — APPROVED** [review]
Great job taking it to the finish line. I left one comment where I’m not entirely sure whether the space at the end of translation should get removed. Should we enable the same rules inside ESLint config that we publish on npm in the follow up PR?

File: packages/block-library/src/post-navigation-link/edit.js (hunk 1 above)
Comment (swissspidy) [reply packages/block-library/src/post-navigation-link/edit.js:52]: Indeed. Looks like it's needed.

Fixed now and fixed the translator comment as well.

**Comment by swissspidy** [comment]
> Should we enable the same rules inside ESLint config that we publish on npm in the follow up PR?

Yes that makes sense to me.

The rules were added in #38225 for React Native / Android in mind, but as you can see here they're useful in general.

Apparently I asked about this before in https://github.com/WordPress/gutenberg/issues/19777#issuecomment-1173946209 but never got an answer why this wasn't enabled by default everywhere.

**Comment by gziolo** [comment]
> Apparently I asked about this before in https://github.com/WordPress/gutenberg/issues/19777#issuecomment-1173946209 but never got an answer why this wasn't enabled by default everywhere.

Right, I see it now. Yes, let's add it in a follow-up and land the changes for Gutenberg for now 🚀

**Pull request merged**